assets/js/modules/analytics-4/datastore/audience-settings.js

```diff
@@ -0,0 +1,282 @@
+	baseName: 'saveAudienceSettings',
+	controlCallback: ( settings ) =>
+		API.set( 'modules', 'analytics-4', 'save-audience-settings', {
+			settings,
+		} ),
+	reducerCallback: fetchStoreReducerCallback,
+	argsToParams: ( settings ) => settings,
+	validateParams: validateAudienceSettings,
+} );
+
+const fetchSyncAvailableAudiencesStore = createFetchStore( {
+	baseName: 'syncAvailableAudiences',
+	controlCallback: () =>
+		API.set( 'modules', 'analytics-4', 'sync-audiences' ),
+	reducerCallback: createReducer( ( state, audiences ) => {
+		if ( ! state.audienceSettings ) {
+			state.audienceSettings = {};
+		}
+
+		state.audienceSettings.availableAudiences = audiences;
+	} ),
+} );
+
+// Actions
+const SET_AVAILABLE_AUDIENCES = 'SET_AVAILABLE_AUDIENCES';
+const SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY =
+	'SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY';
+
+const baseInitialState = {
+	audienceSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Sets the available audiences.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Array} availableAudiences Available audience resource names.
+	 * @return {Object} Redux-style action.
+	 */
+	setAvailableAudiences( availableAudiences ) {
+		invariant(
+			Array.isArray( availableAudiences ),
+			'Available audiences should be an array.'
+		);
+
+		return {
+			type: SET_AVAILABLE_AUDIENCES,
+			payload: { availableAudiences },
+		};
+	},
+
+	/**
+	 * Sets the user who set up Audience Segmentation.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {number} audienceSegmentationSetupCompletedBy ID for the user who set up Audience Segmentation.
+	 * @return {Object} Redux-style action.
+	 */
+	setAudienceSegmentationSetupCompletedBy(
+		audienceSegmentationSetupCompletedBy
+	) {
+		// Should be an integer.
+		invariant(
+			typeof audienceSegmentationSetupCompletedBy === 'number',
+			'audienceSegmentationSetupCompletedBy by should be an integer.'
+		);
+
+		return {
+			type: SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY,
+			payload: { audienceSegmentationSetupCompletedBy },
+		};
+	},
+
+	/**
+	 * Saves the audience settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} settings Audience settings to save.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	saveAudienceSettings: createValidatedAction(
+		( settings ) => {
+			validateAudienceSettings( settings );
+		},
+		function* ( settings ) {
+			yield clearError( 'saveAudienceSettings', [] );
+
+			const { response, error } =
+				yield fetchSaveAudienceSettingsStore.actions.fetchSaveAudienceSettings(
+					settings
+				);
+
+			if ( error ) {
+				yield receiveError( error, 'saveUserAudienceSettings', [] );
+			}
+
+			return { response, error };
+		}
+	),
+};
+
+const baseResolvers = {
+	*getAvailableAudiences() {
+		const registry = yield commonActions.getRegistry();
+		const { select, dispatch } = registry;
+
+		const audiences = select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
+
+		if ( audiences === undefined ) {
```

Comment:
@ankitrox, I've just realised this condition is wrong, it should be testing for `null` like the current version does.

```js
		if ( audiences === null ) {
```

We can address this in the PR for #8888, I'll make note there.